merge the dev to master to 2.0 version release

jianhong opened the pull request:
<!--
# nf-core/hicar pull request

Many thanks for contributing to nf-core/hicar!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

Remember that PRs should be made against the dev branch, unless you're preparing a pipeline release.

Learn more about contributing: [CONTRIBUTING.md](https://github.com/nf-core/hicar/tree/master/.github/CONTRIBUTING.md)
-->

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/hicar/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, also make a PR on the nf-core/hicar _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

jianhong committed:
update README.md for Zenodo DOI.

jianhong committed:
Update README.md

Co-authored-by: James A. Fellows Yates <jfy133@gmail.com>

jianhong committed:
update the version number to dev.

jianhong committed:
Update README.md

Co-authored-by: James A. Fellows Yates <jfy133@gmail.com>

jianhong committed:
update changelog.md

jfy133 committed:
Update CHANGELOG.md

jfy133 committed:
Merge pull request #69 from jianhong/master

update the zenodo doi.

nf-core-bot committed:
Template update for nf-core/tools version 2.4

jianhong committed:
add support for MseI.

jianhong committed:
Merge remote-tracking branch 'upstream/dev' into dev

jianhong committed:
Merge remote-tracking branch 'upstream/nf-core-template-merge-2.4' into dev

jianhong committed:
fix the issue of multiqc_config file

jianhong committed:
fix a issue in README.

jianhong committed:
add TAD, AB compartments, APA analysis

jianhong committed:
change the pipeline to use different tools

jianhong committed:
update workflow image.

jianhong committed:
re-orgnize the juicer tools modules.

jianhong committed:
add multiple scripts for homer

jianhong committed:
update scripts for hicdcplus

jianhong committed:
add filter condition for prepare_counts.

jianhong committed:
fix a typo in homer_compartments

nf-core-bot committed:
Template update for nf-core/tools version 2.5

nf-core-bot committed:
Template update for nf-core/tools version 2.5.1

nf-core-bot committed:
Template update for nf-core/tools version 2.6

jianhong committed:
change the output folder for test_hipeak.conf

jianhong committed:
fix a bug if juicer_tools_jar is false, there is no JUICER_PRE.out

jianhong committed:
Merge remote-tracking branch 'upstream/nf-core-template-merge-2.6' into dev2

jianhong committed:
fix lint

jianhong committed:
prettier write

jianhong committed:
fix the space issue for enzyme_cut.nf

jianhong committed:
fix multiple lint issue

jianhong committed:
reformat modules.config;
add merge interactions module;

jianhong committed:
update compartment, tad, apa tools.

jianhong committed:
add hicdcplus interaction caller.

jianhong committed:
update more differential methods.
update more v4c.
update more loop calling.

jianhong committed:
update v4c and circos plot.

jianhong committed:
fix an issue of hicdcplus callloops

jianhong committed:
output channels for debug.

jianhong committed:
add TFEA

jianhong committed:
fix multiple bugs in hicdcplus.

jianhong committed:
update homer install.nf

jianhong committed:
update homer install.nf

nf-core-bot committed:
Template update for nf-core/tools version 2.7.1

jianhong committed:
Merge remote-tracking branch 'upstream/nf-core-template-merge-2.7.1' into dev2

jianhong committed:
update documentation and modules.

jianhong committed:
remove local_modules test.

jianhong committed:
update documentations

jianhong committed:
fix the white space issue.

jianhong committed:
fix a typo in eigenvector.nf

jianhong committed:
add peakachu.

jianhong committed:
run python black

jianhong committed:
fix the read_csv error for MAPS.py

jianhong committed:
update annotations from 5k,10K bins to R2 peaks; fix peakachu.

jianhong committed:
remove the usage.html

jianhong committed:
chmod peakachu

jianhong committed:
fix peakachu.

nf-core-bot committed:
Template update for nf-core/tools version 2.7.2

jianhong committed:
adjust the output folder

jianhong committed:
add tools input check.

jianhong committed:
Merge remote-tracking branch 'upstream/nf-core-template-merge-2.7.2' into dev2

jianhong committed:
fix prettier check

jianhong committed:
fix the duplicated imported module issue.

jianhong committed:
update documentation

jianhong committed:
fix the bugs in atacseqtfea.nf

jianhong committed:
fix the chromosome style for homer tfea analysis;
fix the bug in homer compartments caller.

jianhong committed:
update documentations.

jianhong committed:
update documentation.

jianhong committed:
prettier

jianhong committed:
try to fix a bug in peakachu model.nf

jianhong committed:
fix the difference in seqlevels style when annotation by chippeakanno.

jianhong committed:
handle the errors for chippeakanno.nf

jianhong committed:
Update assets/multiqc_config.yml

Co-authored-by: James A. Fellows Yates <jfy133@gmail.com>

jianhong committed:
fix a bug in recenterpeak.nf for bed format file.

jianhong committed:
Merge branch 'dev2' of github.com:jianhong/hicar into dev2

jianhong committed:
Update docs/usage.md

Co-authored-by: James A. Fellows Yates <jfy133@gmail.com>

jianhong committed:
Update docs/usage.md

Co-authored-by: James A. Fellows Yates <jfy133@gmail.com>

jianhong committed:
Update docs/usage.md

Co-authored-by: James A. Fellows Yates <jfy133@gmail.com>

jianhong committed:
update documentation.

jianhong committed:
add license to customized scripts.

jianhong committed:
rename the tools.

jianhong committed:
move the tools check step to nextflow schema.
remove the java memory parameter.
remove the unused scripts.

jianhong committed:
remove the ifEmpty for versions.
replace combine by join

jianhong committed:
remove if statement for juicer_tools_jar check.

jianhong committed:
remove homer interaction caller.

jianhong committed:
Update subworkflows/local/interaction_caller/maps_multienzyme.nf

Co-authored-by: James A. Fellows Yates <jfy133@gmail.com>

jianhong committed:
Update workflows/hicar.nf

Co-authored-by: James A. Fellows Yates <jfy133@gmail.com>

jianhong committed:
removed un-used code.

jianhong committed:
add doi to nextflow.config file.

jianhong committed:
mark todo for mqc channel.

jianhong committed:
change local cooler modules to nf-core cooler modules.

jianhong committed:
Update subworkflows/local/interactions.nf

Co-authored-by: James A. Fellows Yates <jfy133@gmail.com>

jianhong committed:
Update subworkflows/local/tads.nf

Co-authored-by: James A. Fellows Yates <jfy133@gmail.com>

jianhong committed:
fix typo of hicexplorer

jianhong committed:
prittier fix the format of nextflow_schema.json

jianhong committed:
reformat peakachu_score_genome.py by black.

jianhong committed:
Update subworkflows/local/tads.nf

Co-authored-by: James A. Fellows Yates <jfy133@gmail.com>

jianhong committed:
add chicQualityControl to filter reference points.

jianhong committed:
fix the channel re-use issue for juicebox.

jianhong committed:
fix hpc usage documentation for slurm job queue size.

jianhong committed:
download the scripts of MAPS to bin folder;
fix the wrong chr2 columns in MAPS calling;
update the juicer tools to release V3.

jianhong committed:
black bin fix for maps scripts.

jianhong committed:
skip error for juicer compartment caller.

jianhong committed:
change the output folder of MAPS from MAPSpeak to maps;
export juicer normalization method as a parameter;
generate bigwig file for juicer compartment results.

jianhong committed:
fix the output folder for juicer bigwig file.
export bigwig file for homer compartment caller.

jianhong committed:
update ChangeLogs.

jianhong committed:
update ChangeLog.md

jianhong committed:
update pythonBlack

jianhong committed:
fix a typo introduced by rename the parameter for CHICSIGNIFICANTINTERACTIONS

jianhong committed:
fix the out of range issue for compartments caller when generate bigwig files.

jianhong committed:
fix the bedgraph file format output by find_hic_compartments.nf.

jianhong committed:
try to fix the issue of End coordinate bigger than chromosome size.

jianhong committed:
fix the error introduced by assign mismatched seqnames for trimming seqlengths.

jianhong committed:
sort the bedgraph before export by trimbedgraph.nf

jianhong committed:
add scale factor to atac reads coverage cal.

jianhong committed:
update changelog.

jianhong committed:
adjust callpeak resource;
set GET_SCALE output to false;

jianhong committed:
fix the coordinates bug in diffhicar by edger

jianhong committed:
try to fix the sort issue for trimbedgraph.

jianhong committed:
fix the ambiguous redirect error for different version of wc

jianhong committed:
fix the bug of ATAC coverage introduced by scale coverage.

jianhong committed:
fix the typo

jianhong committed:
try to fix the sortbed issue for trimbegraph.nf

jianhong committed:
fix the negative width value for trimbedgraph.

jianhong committed:
update the circos plot.

jianhong committed:
fix the indent issue.

jianhong committed:
fix the bug that peakachu_score_genome only output one chromosome.
update circos.conf.

jianhong committed:
update reads_summary table to include unmapped/multiple mapped info.

jianhong committed:
add the possibility to subsample to balance the input reads

jianhong committed:
update circos plot
update hicexplorer tad plot

jianhong committed:
update circos plots for homer

jianhong committed:
keep standard chromosome only for hicdcplus.

jianhong committed:
change the cutadapt error tolerance from 0 to 0.15

FriederikeHanssen committed:
update paths

ewels committed:
Merge pull request #81 from FriederikeHanssen/aws_paths

update paths

jianhong committed:
fix the issue of dot in group name for APA plot.

jianhong committed:
replace the dots by '_' in the samples name

jianhong committed:
Prettier check

jianhong committed:
fix the bug in merge process if no interaction called.

jianhong committed:
fix the sort method for juicer pre.

jianhong committed:
change the default normalization method of juicer tools from KR to SCALE.

jianhong committed:
update cooltools version;
fix a bug in chromsome name filter for hipeak.

jianhong committed:
fix the bug that circos can not handle emsembl style chromosome names.

jianhong committed:
fix the bug that circos can not handle ensembl style chromosome names.

jianhong committed:
add more annotation plots.

jianhong committed:
fix the spaces in last update.

jianhong committed:
adjust parameters for cooltools/insulation

jianhong committed:
loosing hipeak calling condition.

nf-core-bot committed:
Template update for nf-core/tools version 2.8

jianhong committed:
merge  upstream/nf-core-template-merge-2.8

jianhong committed:
fix the issue `TypeError: float() argument must be a string or a real number, not 'collections.OrderedDict'` introduced by multiQC.

jianhong committed:
update modules;
comment out registry to quay.io until the bugs fixed for gunzip and 
untar.

jianhong committed:
adjust the pipeline controlers;
update the docker.registry.

maxulysse committed:
Merge pull request #84 from jianhong/dev

rename Dev2 to Dev

jianhong committed:
update version number.

jianhong committed:
nf-core lint   --fix files_unchanged

jianhong committed:
Revert "nf-core lint   --fix files_unchanged"

This reverts commit 180716f62f8ea9cac73827a151e1a1652ca8c44f.

jianhong committed:
remove the memory setting for samtools sort until the bug be fixed.

jianhong committed:
try .8 times memory for samtools_sort

jianhong committed:
not fix, leave it unsolved.

jianhong committed:
temporary fix until the official update.

jianhong committed:
use local samtools/sort instead of nf-core/samtools/sort to avoid the memory issue;
remove the modules config for apa/*

jianhong committed:
update the change log file.

sateeshperi committed:
Merge pull request #85 from jianhong/dev

Prepare for release v2

SPPearce commented on assets/multiqc_config.yml:
```diff
@@ -3,9 +3,164 @@ report_comment: >
+  pairs_reads_proportion:
+    parent_id: "pairs_qc"
+    section_name: "Proportion of read orientation versus genomic separation"
+    description: "Contact proportion of reads are shown, stratified by read orientation. Good four cutter samples would show no bias in propotion >3kb(10^3.5)."
```
```suggestion
    description: "Contact proportion of reads are shown, stratified by read orientation. Good four cutter samples would show no bias in proportion >3kb(10^3.5)."
```

SPPearce replied:
What does "four cutter" mean here?

SPPearce commented on assets/multiqc_config.yml:
```diff
@@ -3,9 +3,164 @@ report_comment: >
+      id: "pairs_reads_proportion_vs_distance"
+      title: "Reads distance vs proportions"
+      xlab: "distance (kb)"
+      ylab: "Propotion"
```
```suggestion
      ylab: "Proportion"
```

SPPearce commented on assets/test_multi_samplesheet.csv:
```diff
@@ -1,9 +1,9 @@
-WT,2,https://raw.githubusercontent.com/nf-core/test-datasets/hicar/data/genomics/homo_sapiens/T2/WT_rep2_R1.fastq.gz,https://raw.githubusercontent.com/nf-core/test-datasets/hicar/data/genomics/homo_sapiens/T2/WT_rep2_R2.fastq.gz,,
-KD,1,https://raw.githubusercontent.com/nf-core/test-datasets/hicar/data/genomics/homo_sapiens/T2/KD_rep1_R1.fastq.gz,https://raw.githubusercontent.com/nf-core/test-datasets/hicar/data/genomics/homo_sapiens/T2/KD_rep1_R2.fastq.gz,,
-KD,2,https://raw.githubusercontent.com/nf-core/test-datasets/hicar/data/genomics/homo_sapiens/T2/KD_rep2_R1.fastq.gz,https://raw.githubusercontent.com/nf-core/test-datasets/hicar/data/genomics/homo_sapiens/T2/KD_rep2_R2.fastq.gz,,
+WT.dot,1,https://raw.githubusercontent.com/nf-core/test-datasets/hicar/data/genomics/homo_sapiens/T1/WT_rep1_R1.fastq.gz,https://raw.githubusercontent.com/nf-core/test-datasets/hicar/data/genomics/homo_sapiens/T1/WT_rep1_R2.fastq.gz,,
```
Why is this `WT.dot` instead of just `WT`? Is that intentional?

edmundmiller replied:
I wonder if it would break stuff, or are the groups separated using `.` ?

SPPearce commented on bin/create_hicar_track_ini.py:
```diff
@@ -0,0 +1,70 @@
+file_type = domains
+border_color = black
+color = none
+# the tads are overlay over the hic-matrix
```
```suggestion
# the tads are overlaid over the hic-matrix
```

SPPearce commented on modules/local/hipeak/assign_type.nf:
```diff
@@ -47,7 +47,7 @@ process ASSIGN_TYPE {
     OUTPUT = "."
     GROUP_ID = "$meta.id"
-    COUNT_CUTOFF = 12
+    COUNT_CUTOFF = 3
```
Is this change in the CHANGELOG?

SPPearce commented on modules/local/hipeak/prepare_counts.nf:
```diff
@@ -67,6 +67,7 @@ process PREPARE_COUNTS {
     OUTPUT <- "counts.${meta.id}.${chrom1}.rds"
     NCORE <- as.numeric("$task.cpus")
     SNOW_TYPE <- "SOCK"
+    COUNTS_FILTER <- 1
```
Has this change been noted in the CHANGELOG?

SPPearce commented on CHANGELOG.md:
```diff
@@ -3,10 +3,42 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## v2.0.0 - [03/15/2023]
```
Date not correct

mirpedrol commented on bin/check_samplesheet.py:
You should add the license in the header of this script. From the [docs](https://nf-co.re/docs/contributing/pipeline_release_review_guidelines#do-local-code-and-modules): `Local script licenses if written for the pipeline should not GPL, but ideally MIT (matching with pipeline code, upon which is assumed if not otherwise specified)`

mirpedrol commented on bin/feature_frag2bin.py:
Not mandatory for this release, but I think it would be good if you create nf-core/modules for these scripts which come from a tool and aren't customized. It would reduce the amount of custom bin files and local modules and will ease the review process and maintenance of the code :)

jianhong replied:
I also hope so. However, after several round of communication, the authors are busy in other stuff and hard to meet the timeline. I will put it as TODO list.

mirpedrol commented on docs/output.md:
```diff
@@ -14,16 +14,21 @@ The pipeline is built using [Nextflow](https://www.nextflow.io/) and processes d
-- [MAPSpeak](#chromatin-interactions) - The chromatin interactions determined by MAPS.
+- [A/B Compartments](#Compartments) - Called A/B compartments.
+- [TADs](#TADs) - Called topologically associating domains.
+- [Interactions](#chromatin-interactions) - The chromatin interactions determined by MAPS/HiCDCPlus or other tools.
```
```suggestion
- [A/B Compartments](#compartments) - Called A/B compartments.
- [TADs](#tads) - Called topologically associating domains.
- [Interactions](#chromatin-interactions) - The chromatin interactions determined by MAPS/HiCDCPlus or other tools.
```
For the links to the different sections to work, they need to be written in lower case here and replacing all spaces by dashes

mirpedrol commented on docs/output.md:
```diff
@@ -200,18 +246,35 @@ It will remove the systematic bias by accessibility.
-Differential analysis for the loops by [edgeR](https://pubmed.ncbi.nlm.nih.gov/19910308/).
+Differential analysis for the loops by [edgeR](https://pubmed.ncbi.nlm.nih.gov/19910308/),
+[diffHic](https://doi.org/10.1186/s12859-015-0683-0) or simply set operation will be put in the folder.
+The subfolder names are the tools used in differentail analysis and in each subfolder,
```
```suggestion
The subfolder names are the tools used in differential analysis and in each subfolder,
```

jianhong replied:
The typo of `differential` will be fixed.

mirpedrol commented on docs/usage.md:
```diff
@@ -61,40 +63,127 @@ TREATMENT,3,AEG588A6_S6_L004_R1_001.fastq.gz,AEG588A6_S6_L004_R2_001.fastq.gz,,
+
+### Call interactions/loops
+
+Chromatin loops (or significant interactions), represent two inter/intra chromosome regions that interact at a high frequency with one another (high reads density in sequence data). Different from HiC, HiCAR data are biased with one ends or both ends in the open chromatin. The [`MAPS`](https://github.com/ijuric/MAPS) are designed to remove the this kind of biases introduced by the ChIP or Tn5-tra ...
```
```suggestion
Chromatin loops (or significant interactions), represent two inter/intra chromosome regions that interact at a high frequency with one another (high reads density in sequence data). Different from HiC, HiCAR data are biased with one ends or both ends in the open chromatin. The [`MAPS`](https://github.com/ijuric/MAPS) are designed to remove this kind of biases introduced by the ChIP or Tn5-transposition procedure. However, many tools are hesitated to introduce this kind of model-based analysis for interaction analysis since high frequency interactions must happened within the highly opened chromatin regions. Here `nf-core/hicar` provide multiple choices for interactions calling. Available tools are 'MAPS', ['HiC-DC+'](https://doi.org/10.1038/s41467-021-23749-x) and ['peakachu'](https://doi.org/10.1038/s41467-020-17239-9).
```

SPPearce replied:
"However, many tools are hesitated to introduce this kind of model-based analysis" should probably be "However, many tools are hesitant to".
"must happened" should be "must happen".
"provide multiple choices" should be "provides multiple choices".

jianhong replied:
Ha, shame. Thank you for correct the grammars.

mirpedrol commented on modules/local/atacreads/getscale.nf:
```diff
@@ -0,0 +1,31 @@
+
+    output:
+    tuple val(meta), val(scale)  , emit: scale
+    path "versions.yml"                               , emit: versions
```
Could you align the commas?

mirpedrol commented on modules/local/atacreads/getscale.nf:
```diff
@@ -0,0 +1,31 @@
+    scale = min_counts/counts.toInteger()
+    def prefix   = task.ext.prefix ? "${meta.id}${task.ext.prefix}" : "${meta.id}"
+    """
+    awk 'BEGIN { print $min_counts/$counts }'
```
Is this used? Maybe you can use the [`env`](https://www.nextflow.io/docs/latest/process.html#output-type-env) type to retrieve the value instead?

jianhong replied:
I will try to rewrite this module.

mirpedrol commented on modules/local/bioc/atacseqtfea.nf:
```diff
@@ -0,0 +1,192 @@
+process BIOC_ATACSEQTFEA {
+    label 'process_high_memory'
+    label 'process_long'
+    label 'process_single'
```
You can override the required parts of a label in a config file. https://nf-co.re/usage/configuration#tuning-workflow-resources

SPPearce replied:
I think this is quite a good way of doing it, but I think it should be the other way round, e.g. single, long, high_memory.

mirpedrol commented on modules/local/bioc/bedpe2bed.nf:
```diff
@@ -0,0 +1,63 @@
+        'quay.io/biocontainers/bioconductor-rtracklayer:1.50.0--r40h7f5ccec_2' }"
+
+    input:
+    tuple val(meta), path(fname)
```
is `fname` used?

jianhong replied:
Yes it is used in the scripts.

mirpedrol commented on modules/local/bioc/chippeakanno.nf:
```diff
@@ -1,33 +1,35 @@
+    tuple val(foldername), path("${prefix}/*"), emit: anno
+    tuple val(foldername), path("${prefix}/**.anno.csv"), emit: csv
+    path "${prefix}/*.png", optional:true, emit: png
     path "versions.yml"                       , emit: versions
```
could you align the emit?

mirpedrol commented on modules/local/bioc/diffhic.nf:
```diff
@@ -0,0 +1,267 @@
+    val long_bedpe_postfix
+
+    output:
+    tuple val(bin_size), path("${prefix}/*")               , emit: diff
```
could you align the emit?

mirpedrol commented on modules/local/bioc/diffset.nf:
```diff
@@ -0,0 +1,93 @@
+    container "${ workflow.containerEngine == 'singularity' &&
+                    !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/r-upsetr:1.0.3--r3.3.2_1' :
+        'quay.io/biocontainers/r-upsetr:1.0.3--r3.3.2_1' }"
```
The version in conda is not the same as for docker % singularity

mirpedrol commented on modules/local/bioc/diffset.nf:
```diff
@@ -0,0 +1,93 @@
+    val long_bedpe_postfix
+
+    output:
+    tuple val(bin_size), path("${prefix}/*")               , emit: diff
```
align emit :)

mirpedrol commented on modules/local/bioc/atacseqtfea.nf:
```diff
@@ -0,0 +1,192 @@
+process BIOC_ATACSEQTFEA {
+    label 'process_high_memory'
```
please add a tag

mirpedrol commented on modules/local/atacreads/mergepeak.nf:
```diff
@@ -1,7 +1,7 @@
 process MERGE_PEAK {
     label 'process_medium'
```
please add a tag

mirpedrol commented on modules/local/atacreads/atacqc.nf:
```diff
@@ -1,7 +1,7 @@
 process ATACQC {
     label 'process_medium'
```
please add a tag

mirpedrol commented on modules/local/bioc/trackviewer.nf:
```diff
@@ -96,23 +97,39 @@ process BIOC_TRACKVIEWER {
 
     # read the signals
-    cools <- dir(".", "mcool\$", full.names = TRUE, recursive = TRUE)
+    cools <- dir(".", "cool\$", full.names = TRUE, recursive = TRUE)
```
the input variable is called `cools`, not `cool`, might be an error? `cools` is overridden in this line

mirpedrol commented on modules/local/cooltools/plotnpz.nf:
```diff
@@ -0,0 +1,65 @@
+
+    input:
+    tuple val(meta), path(npz)
+    val format
```
this could be provided through args

mirpedrol commented on modules/local/cooltools/virtual4c.nf:
```diff
@@ -0,0 +1,56 @@
+
+    input:
+    tuple val(bin_size), path(cool), path(viewpoint)
+    val v4c_max_events
```
please provide through args

nf-core-bot committed:
Template update for nf-core/tools version 2.9

mirpedrol reviewed (commented):
Still halfway through the review, but I am leaving some comments already, will continue next week :)

mirpedrol commented on modules/local/hicdcplus/callloops.nf:
```diff
@@ -0,0 +1,147 @@
+    tuple val(meta), val(meta.bin), path("*.interactions.bedpe")     , emit: interactions
+    tuple val(meta), val(meta.bin), path("*_hicdcplus_result.txt.gz"), emit: full_results
+    tuple val(meta), val(meta.bin), path("*.interactions.txt")       , emit: filtered_res
+    path "versions.yml"                          , emit: versions
```
Align emit

mirpedrol commented on modules/local/hicexplorer/chicqualitycontrol.nf:
```diff
@@ -0,0 +1,33 @@
+    conda "bioconda::hicexplorer=3.7.2 cleanlab=1.0.1"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/hicexplorer:3.7.2--pyhdfd78af_1' :
+        'quay.io/biocontainers/hicexplorer:3.7.2--pyhdfd78af_1' }"
```
You should remove the `quay.io/` from all modules as we are using a registry now, which is `quay.io` by default. See `docker.registry` or `signularity.registry` in `nextflow.config` file.
```suggestion
        'biocontainers/hicexplorer:3.7.2--pyhdfd78af_1' }"
```

mirpedrol commented on modules/local/hicexplorer/hicfindtads.nf:
```diff
@@ -0,0 +1,50 @@
+    def bin_size = meta.bin.toInteger()
+    def mDepth = [bin_size * 3, bin_size * 10]
+    def mDepthArg = ['--minDepth', '--maxDepth']
+    args = args.tokenize()
+    for(i=0; i<2; i++){
+        idx = args.indexOf(mDepthArg[i])
+        if(idx>=0){
+            mDepth[i] = args[idx + 1]
+            args.remove(idx+1)
+            args.remove(idx)
+        }
+    }
+    args = args.join(' ')
```
Not sure what this code is doing, would it be possible to simplify this by providing all arguments trhough `modules.config`?

edmundmiller replied:
A comment of what it's trying to achieve would be helpful as well.

mirpedrol commented on modules/local/hichipper/hichipper.nf:
```diff
@@ -0,0 +1,31 @@
+        'quay.io/biocontainers/hichipper:0.7.7--py_1' }"
+
+    input:
+    tuple val(meta), path(hicpro), path(peak), path(yaml)
```
is `hicpro` used?

mirpedrol commented on modules/local/homer/install.nf:
```diff
@@ -0,0 +1,38 @@
+    val genome
+
+    output:
+    stdout  emit: output
```
A comma is missing

mirpedrol commented on modules/local/re_cut.nf:
```diff
@@ -0,0 +1,26 @@
+    val enzyme
+
+    output:
+    stdout emit: site
```
A comma is missing

mirpedrol commented on modules/local/re_cut.nf:
```diff
@@ -0,0 +1,26 @@
+    container "${ workflow.containerEngine == 'singularity' &&
+                    !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/biopython:1.78' :
+        'quay.io/biocontainers/biopython:1.78' }"
```
container versions don't match

mirpedrol commented on modules/local/re_cut.nf:
```diff
@@ -0,0 +1,26 @@
+process RE_CUTSITE {
+    label 'process_low'
```
please add a tag

mirpedrol commented on modules/local/re_cut.nf:
```diff
@@ -0,0 +1,26 @@
+        python: \$(echo \$(python --version) | sed 's/Python //')
+    END_VERSIONS
+
+    restriction_enzyme_cutsite.py $enzyme
```
It is better to run the command before emitting the version

mirpedrol commented on modules/local/samtools/sort/main.nf:
Not sure why this module is now a local module instead of a nf-core module?

mirpedrol commented on nextflow.config:
```diff
@@ -35,22 +34,54 @@ params {
+    skip_trackhub              = false
+    create_virtual_4c          = false
+    skip_circos                = false
+    skip_maps_signal           = true
```
default booleans should be `false`

mirpedrol commented on nextflow.config:
```diff
@@ -134,66 +183,112 @@ try {
     }
     docker {
         docker.enabled         = true
+        docker.registry        = 'quay.io'
```
registries should be defined outside of the profiles, otherwise some runs will fail, for example in Tower. This is fixed in the nf-core template version 2.9

mirpedrol commented on subworkflows/local/compartments.nf:
```diff
@@ -0,0 +1,84 @@
+workflow COMPARTMENTS {
+    take:
+    matrix                                       // tuple val(meta), path(cool/tagdir)
+    resolution
+    additional_param
+    cool_bin
```
Could you specify the content of these channels too?

mirpedrol commented on subworkflows/local/compartments.nf:
```diff
@@ -0,0 +1,84 @@
+
+    emit:
+    compartments    = ch_compartments      // channel: [ meta, [compartments] ]
+    circos          = ch_circos_files
+    igv             = ch_track_files
+    versions        = ch_versions          // channel: [ versions.yml ]
+    mqc             = ch_multiqc_files
```
The same for the output channels, please specify the content

mirpedrol commented on subworkflows/local/coveragescale.nf:
I think this should be a module, not a subworkflow, as it only uses a single module.

mirpedrol commented on subworkflows/local/differential_analysis.nf:
```diff
@@ -0,0 +1,79 @@
+
+    main:
+    ch_versions             = Channel.empty()
+    ch_multiqc_files        = Channel.empty() // TODO
```
What does this TODO indicate?

mirpedrol commented on subworkflows/local/hipeak.nf:
```diff
@@ -1,80 +1,78 @@
-    stats        = stats                        // channel: [ path(stats) ]
-    diff         = diff                         // channel: [ path(diff) ]
-    versions     = ch_version                   // channel: [ path(version) ]
+    fragmentPeak    = R1_PEAK.out.peak
+    bed4tfea        = HI_PEAK.out.bed4tfea
+    circos          = ch_circos_files
+    igv             = ch_track_files
+    anno            = ch_annotation_files
+    versions        = ch_versions          // channel: [ versions.yml ]
+    mqc             = ch_multiqc_files
```
Please specify the content of the channels in a comment

mirpedrol commented on subworkflows/local/interaction_caller/maps_peak.nf:
```diff
@@ -0,0 +1,57 @@
+    //create parameter table
+    //input=val(meta), val(bin_size), path(macs2), path(long_bedpe), path(short_bed), path(background)
+    //maps from bedpe
+    ch_version = MAPS_MAPS(reads, params.long_bedpe_postfix, params.short_bed_postfix).versions
```
It would be more clear if we first run the module and then assign the version to the channel.

mirpedrol commented on subworkflows/local/interactions.nf:
```diff
@@ -0,0 +1,96 @@
+
+    main:
+    ch_versions             = Channel.empty()
+    ch_multiqc_files        = Channel.empty() // TODO
```
TODO?

mirpedrol commented on subworkflows/local/interactions.nf:
```diff
@@ -0,0 +1,96 @@
+
+workflow INTERACTIONS {
+    take:
+    matrix
+    peaks
+    additional_param
```
Please specify the content of the channels

mirpedrol commented on subworkflows/local/tads_caller/homer.nf:
Can you directly use the module? It doesn't look like a subworkflow with only one module

mirpedrol commented on subworkflows/local/tfea.nf:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    emit:
+    versions        = ch_versions          // channel: [ versions.yml ]
```
Does it only emit the versions?

mirpedrol commented on subworkflows/local/tfea/homer.nf:
```diff
@@ -0,0 +1,36 @@
+
+
+    emit:
+    versions     = ch_versions                  // channel: [ path(version) ]
```
Does it only emit the versions?

mirpedrol commented on workflows/hicar.nf:
```diff
@@ -72,30 +90,37 @@ ch_make_maps_runfile_source  = file(params.make_maps_runfile_source,
 //
 include { CHECKSUMS } from '../modules/local/checksums'
-include { DIFFHICAR } from '../modules/local/bioc/diffhicar'
+include { HOMER_INSTALL } from '../modules/local/homer/install'
```
Can you align the paths?

mirpedrol commented on workflows/hicar.nf:
```diff
@@ -231,227 +286,389 @@ workflow HICAR {
-        ch_multiqc_files = ch_multiqc_files.mix(BIOC_CHIPPEAKANNO_MAPS.out.png.collect().ifEmpty([]))
-        if(params.virtual_4c){
-            BIOC_CHIPPEAKANNO_MAPS.out.csv
+        BIOC_CHIPPEAKANNO(ch_annotation_files.combine(ATAC_PEAK.out.mergedpeak), PREPARE_GENOME.out.gtf, params.maps_3d_ext)
```
As a suggestion, it is easier to understand if you add each input in a new line and use the appropriate indentation.

SPPearce commented on docs/usage.md:
````diff
@@ -31,6 +31,8 @@ CONTROL,1,AEG588A1_S1_L003_R1_001.fastq.gz,AEG588A1_S1_L003_R2_001.fastq.gz
 CONTROL,1,AEG588A1_S1_L004_R1_001.fastq.gz,AEG588A1_S1_L004_R2_001.fastq.gz
 ```
 
+Please Note that the dots ('.') and space (' ') in the group names will be replaced by '\_' to avoid the potential issues of python scripts. In the pipeline, the group names will be used as prefix of filenames. Many tools developed in python will consider the first dot as the sign of file extension and this will throw errors when the program does not expected the extra dots in the filenames. If y ...
````
```suggestion
Please Note that the dots ('.') and space (' ') in the group names will be replaced by '\_' to avoid potential issues in python scripts. In the pipeline, the group names will be used as the prefix for filenames. Many tools developed in python will consider the first dot as the sign of file extension and this will throw errors when the program does not expected the extra dots in the filenames. If you want to keep consistent of file naming system with the group names, please consider to replace all the dots and spaces in the samplesheet.
```

SPPearce commented on docs/usage.md:
```diff
@@ -61,40 +63,127 @@ TREATMENT,3,AEG588A6_S6_L004_R1_001.fastq.gz,AEG588A6_S6_L004_R2_001.fastq.gz,,
+On a large scale, the arrangement of chromosomes are organised into two compartments labelled A ("active") and B ("inactive").
+A/B compartment-associated regions are on the multi-Mb scale and correlate with either open and expression-active chromatin ("A" compartments) or closed and expression inactive chromatin ("B" compartments). A compartments tend to be gene-rich, have high GC-content, contain histone markers for active transcription, and usually displace the interior of the nucleus. The regions in compartment A tend ...
+
+A topologically associating domain (TAD) is a smaller size genomic region compare to A/B compartments. It is a self-interacting genomic region. Most of the studies indicate TADs regulate gene expression by limiting the enhancer-promoter interaction to each TAD. A number of proteins are known to be associated with TAD formation. The most studied proteins are the protein CTCF and the protein comple ...
```
```suggestion
A topologically associating domain (TAD) is a smaller size genomic region compared to A/B compartments. It is a self-interacting genomic region. Most of the studies indicate TADs regulate gene expression by limiting the enhancer-promoter interaction to each TAD. A number of proteins are known to be associated with TAD formation. The most studied proteins are the protein CTCF and the protein complex cohesin. It has been shown that the TAD boundaries have high levels of CTCF binding and cohesin/lamina shifting edges.
```

SPPearce commented on docs/usage.md:
```diff
@@ -61,40 +63,127 @@ TREATMENT,3,AEG588A6_S6_L004_R1_001.fastq.gz,AEG588A6_S6_L004_R2_001.fastq.gz,,
+
+Here is a short introduction about the tools:
+
+- The [`cooltools`](https://github.com/open2c/cooltools) leverages [`cooler`](https://github.com/open2c/cooler/tree/master/cooler) format to enable flexible and reproducible analysis of high-resolution data. `insulation` tool will be used for TADs calling.
```
```suggestion
- The [`cooltools`](https://github.com/open2c/cooltools) leverages the [`cooler`](https://github.com/open2c/cooler/tree/master/cooler) format to enable flexible and reproducible analysis of high-resolution data. `insulation` tool will be used for TADs calling.
```

SPPearce commented on docs/usage.md:
```diff
@@ -61,40 +63,127 @@ TREATMENT,3,AEG588A6_S6_L004_R1_001.fastq.gz,AEG588A6_S6_L004_R2_001.fastq.gz,,
+
+In downstream, differential analysis available for called interactions. Available tools are Bioconductor packages such as `edgeR`, and `diffhic`, and [`HiCExplorer`](https://hicexplorer.readthedocs.io/en/latest/). We borrowed capture Hi-C analysis pipeline from HiCExplorer to do the differential analysis. Different from `edgeR` and `diffhic` pipeline, HiCExplorer pipeline does not require the rep ...
+
+For annotation, we will use Bioconductor package [`ChIPpeakAnno`](https://bioconductor.org/packages/ChIPpeakAnno/). Please note that, the involved genes are not only distance based annotation. The most of the interaction calling tools are bin-based caller, and the bin size are kilo-base or even more, which make the annotation difficult. For HiCAR data, the R2 reads are Tn5 insertion site of the o ...
```
```suggestion
For annotation, we will use Bioconductor package [`ChIPpeakAnno`](https://bioconductor.org/packages/ChIPpeakAnno/). Please note that, the involved genes are not only distance based annotation. The most of the interaction calling tools are bin-based caller, and the bin size are kilo-base or even more, which make the annotation difficult. For HiCAR data, the R2 reads are Tn5 insertion site of the open chromatin. And most of the R2 reads will be an anchor of annotation for the gene promoters. We will annotate the interactions by the annotation of called ATAC (R2) peaks located within the interaction regions.
```

SPPearce commented on subworkflows/local/v4c/cooltools.nf:
Can you please call `BIOC_SUBSETLOOPS` on its own, then update ch_versions with the output, I find this very hard to read.
Is there a reason why the input channel is just called matrix, and then you take parts of that, as opposed to defining 4 individual parts in one channel?

SPPearce commented on subworkflows/local/tfea/homer.nf:
```diff
@@ -0,0 +1,36 @@
+workflow HOMER_TFEA {
+    take:
+    bed                      // peaks regions [meta, R1/2, [peak]]
+    additional_param         // singals for each tools
```
What is `additional_param` doing here?
Should probably be "signals" not "singals".

SPPearce commented on subworkflows/local/tfea/homer.nf:
The input is just called "bed", when it isn't just a bed file? Could it be the named parts rahter than just "bed" and then you wouldn't have to do as many maps.

SPPearce commented on subworkflows/local/tads_caller/hicexplorer.nf:
```diff
@@ -0,0 +1,34 @@
+
+    emit:
+    tads      = HICEXPLORER_HICFINDTADS.out.tads     // channel: [ val(meta), val(bin), path(domains.bed)]
+    mqc       = ch_multiqc_files                     // channel: [ path(mqc) ]
```
This is empty

SPPearce commented on subworkflows/local/tads_caller/homer.nf:
```diff
@@ -0,0 +1,31 @@
+
+    emit:
+    tads         = ch_tads                      // channel: [ meta, bin_size, path(bedpe) ]
+    mqc          = ch_multiqc_files             // channel: [ path(mqc) ]
```
This is empty

SPPearce commented on subworkflows/local/tads_caller/cooltools.nf:
```diff
@@ -0,0 +1,28 @@
+
+    emit:
+    tads         = ch_tads                      // channel: [ meta, bin_size, path(bedpe) ]
+    mqc          = ch_multiqc_files             // channel: [ path(mqc) ]
```
This is empty.

SPPearce commented on subworkflows/local/tads.nf:
```diff
@@ -0,0 +1,81 @@
+    emit:
+    tads            = ch_tads              // channel: [ meta, bin_size, [TADs] ]
+    circos          = ch_circos_files      // channel: [ meta, [TADs]]
+    igv             = ch_track_files
```
This is empty

SPPearce commented on subworkflows/local/tads.nf:
```diff
@@ -0,0 +1,81 @@
+    tads            = ch_tads              // channel: [ meta, bin_size, [TADs] ]
+    circos          = ch_circos_files      // channel: [ meta, [TADs]]
+    igv             = ch_track_files
+    anno            = ch_annotation_files
```
This is empty

SPPearce reviewed (commented):
I am halfway through reviewing but don't have any more time until Thursday, so leaving my comments so far (at 161/294 files viewed). It is hard to work through so many changes in one go, particularly with template changes and modules changes mixed in.
Generally there are a lot of local modules and subworkflows, some of which could do with going into nf-core, or explaining why you are not using the nf-core version of the module.

jianhong committed:
add license to file `check_samplesheet.py`

jianhong committed:
Update output documentation for missing links by changing to lower case.

jianhong committed:
fix the typo and grammar for the output and usage documentation.

jianhong committed:
remove the unused command line in getscale.

jianhong committed:
fix the alignment style of code.

jianhong committed:
Fix the typo in documentations.

jianhong committed:
remove quay.io from local modules.

jianhong committed:
change parameter `skip_maps_signal` to `create_maps_signal`

jianhong commented:
@mirpedrol @SPPearce I opened a new PR to answer the comments because I can not write to dev version here.

jianhong committed:
fix a type and remove unused codes for getscale.

mirpedrol committed:
Merge pull request #88 from jianhong/dev

answer the comments for the release review

jianhong committed:
merge upstream/nf-core-template-merge-2.9; shift input check to plugin.

jianhong committed:
shift the input check to plugin.

jianhong committed:
add samtools/sort git_sha back.

jianhong committed:
remove the duplicated input plugins.

jianhong committed:
remove unused files; change file exists check by nf-validation.

edmundmiller commented on CHANGELOG.md:
```diff
@@ -3,10 +3,45 @@
+  - fix the chromosome style for homer TFEA analysis and annotation by ChIPpeakAnno;
+  - fix the duplicated imported modules;
+  - fix multiple typos.
+  - use local SAMTOOLS_SORT module;
```
I was going to ask if there were any links to these. But then I remembered this was all done on a fork. Maybe link some commits?

I think we need to add something to the nf-core docs for people to reach out if they're developing on a pipeline.

edmundmiller commented on conf/test.config:
```diff
@@ -20,21 +20,27 @@ params {
 
     // Output folder
-    outdir = 'test_results'
+    outdir = 'results'
```
I believe these were removed because they could cause people using it on cloud to write the results to never write to a bucket. It is just the test config, just a consideration.

edmundmiller commented on conf/test_full.config:
```diff
@@ -18,7 +20,7 @@ params {
 
     // Output folder
-    outdir = 'full_test_results'
+    outdir = 'results'
```
See the above comment, this one probably doesn't need to be set by default, it'll just result in a foot gun that's worse than users have to specify `--outdir results`

edmundmiller commented on modules/local/atacreads/atacqc.nf:
Is there a reason this isn't a script in `bin/`?

edmundmiller commented on modules/local/atacreads/getscale.nf:
Does this need to be a process? Maybe a comment on why this, i.e. so you can spin up a bunch of jobs instead of stressing the head node.

edmundmiller commented on modules/local/bioc/atacseqtfea.nf:
Wondering why this one isn't in `bin/` as well. 

Just a style thing, we can make an issue to track those and get them all into `bin/`.

edmundmiller commented on modules/local/hicdcplus/callloops.nf:
```diff
@@ -0,0 +1,147 @@
+process HICDCPLUS_CALLLOOPS {
```
```suggestion
process HICDCPLUS_CALL_LOOPS {
```

edmundmiller commented on modules/local/hichipper/creat_yaml.nf:
File name should be `yaml.nf` or `create_yaml.nf` currently a typo.

edmundmiller commented on modules/local/hichipper/hichipper.nf:
```diff
@@ -0,0 +1,31 @@
+    script:
+    prefix = task.ext.prefix ?: "${meta.id}_${meta.bin}"
+    """
+    ## step2 run programs
```
```suggestion
    ## step 1 run programs
```

Unless there's a step 1 somewhere else?

edmundmiller commented on modules/local/homer/dump4homer.nf:
Could this be attached to the `find_hic_compartments` module? Just provide both outputs.

edmundmiller commented on modules/local/homer/install.nf:
This module isn't necessary and more importantly causes unintended side effects because of how HOMER installs its stuff.

Not sure if it's actually used, I haven't gotten to the main workflow yet.

I spent a few years trying to wrangle this, until I realized HOMER can just take a fasta 🙃

edmundmiller commented on modules/local/juicer/addnorm.nf:
```diff
@@ -0,0 +1,46 @@
+
+    input:
+    tuple val(meta), path(hic)
+    path hic_tools_jar
```
This should be in a container. I'm guessing it's https://github.com/aidenlab/HiCTools.

Nvm I looked at it and that looks like a nightmare, I guess I'll get to the config and this points at the jar download?

edmundmiller commented on modules/local/maps/maps.nf:
```diff
@@ -2,15 +2,16 @@ process MAPS_MAPS{
     input:
     tuple val(meta), val(bin_size), path(macs2), path(long_bedpe, stageAs: "long/*"), path(short_bed, stageAs: "short/*"), path(background)
-    path make_maps_runfile_source
+    val long_bedpe_postfix
+    val short_bed_postfix
```
Could be args in a config

edmundmiller commented on modules/local/re_cut.nf:
```diff
@@ -0,0 +1,26 @@
+        python: \$(echo \$(python --version) | sed 's/Python //')
+    END_VERSIONS
+
+    restriction_enzyme_cutsite.py $enzyme
```
```suggestion
    restriction_enzyme_cutsite.py $enzyme 

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(echo \$(python --version) | sed 's/Python //')
    END_VERSIONS
```

edmundmiller commented on modules/local/samtools/sort/main.nf:
Could why this is a local module get documented? What'st the reason for not using a patch and the nf-core module? That will help keep this maintainable.

edmundmiller commented on nextflow.config:
```diff
@@ -60,40 +91,53 @@ params {
 
     // Boilerplate options
-    outdir                     = null
+    outdir                     = 'results'
```
```suggestion
    outdir                     = null
```

edmundmiller reviewed (commented):
This is a great effort, and it's obvious you've put a lot of time into this!

It seems there are a couple of repetitive things that I'd like to see addressed (beyond what @mirpedrol and @SPPearce pointed out)

- Inline scripts in Nextflow files. Those should be in `bin/` or in a custom module. Just a nf-core thing, it makes it easier to maintain. I'm guessing you did that to be able to test the nextflow local modules, but since you've written so many custom scripts, I think it might be better to just test them with testthat, or another CLI tester. They're really extensive, so it would be nice to have confidence in them!
- I'd like to see the hicexplorer, homer, etc. modules live in nf-core/modules
- Make an image for those scary java tools 🙈 
- Homer modules need to use fasta input rather than the outside of Nextflow genome install. I'm not sure if it's working as intended as is, it would install the genome in the process' workdir. 
- Clean up subworkflows only outputting versions, and empty outputs. It'll make it more maintainable.

So I don't think this is entirely up to nf-core standards of maintainability, but I think in the name of progress we merge this and maybe tag it with `2.0.0-rc` or something if `2.0.0` is controversial. Also, up to @jianhong and how they want to release it.

Before I would approve this PR, I'd like to see a roadmap tracked in the milestones of these issues to be addressed from this PR (and maybe with a list of modules to move scripts out of, for example). I think those will be easier to address in separate, smaller PRs. 

We should also make sure proper repo access is given to @jianhong, so they can work on branches in this repo and create smaller PRs, so the review process doesn't take multiple months, and so we can review the scientific ideas in the code changes.

jianhong committed:
try to fix an issue for annotation when empty file is input.

jianhong committed:
remove the input_check.nf.

jianhong committed:
simplify the input meta data rearrangement code.

jianhong committed:
Merge pull request #89 from jianhong/dev

merge template 2.9

jianhong committed:
remove output folder in the config files.

jianhong committed:
Merge branch 'nf-core:dev' into dev_remove_output_folder

jianhong committed:
update Changelog.md

jianhong committed:
update Changelog.md

jianhong committed:
fix a issue in if id contains dots.

jianhong committed:
Merge pull request #92 from jianhong/dev

fix a issue in if id contains dots.

jianhong committed:
Merge branch 'nf-core:dev' into dev_remove_output_folder

jianhong committed:
fix the meta issue.

jianhong committed:
improve the meta id related codes.

jianhong committed:
Merge pull request #90 from jianhong/dev_remove_output_folder

Dev remove output folder from config file

jianhong committed:
Merge branch 'nf-core:dev' into dev

jianhong committed:
Merge pull request #93 from jianhong/dev

fix the meta id issue.

jianhong committed:
update the documentation for technique replicate.

JoseEspinosa commented on README.md:
```diff
@@ -22,7 +19,7 @@ The pipeline can also handle the experiment of HiChIP, ChIA-PET, and PLAC-Seq. I
 The pipeline is built using [Nextflow](https://www.nextflow.io), a workflow tool to run tasks across multiple compute infrastructures in a very portable manner. It uses Docker/Singularity containers making installation trivial and results highly reproducible. The [Nextflow DSL2](https://www.nextflow.io/docs/latest/dsl2.html) implementation of this pipeline uses one container per process which mak ...
 
-On release, automated continuous integration tests run the pipeline on a full-sized dataset on the AWS cloud infrastructure. This ensures that the pipeline runs on AWS, has sensible resource allocation defaults set to run on real-world datasets, and permits the persistent storage of results to benchmark between pipeline releases and other analysis sources. The results obtained from the full-sized ...
+On release, automated continuous integration tests run the pipeline on a full-sized dataset on the AWS cloud infrastructure. This ensures that the pipeline runs on AWS, has sensible resource allocation defaults set to run on real-world datasets, and permits the persistent storage of results to benchmark between pipeline releases and other analysis sources.The results obtained from the full-sized  ...
```
```suggestion
On release, automated continuous integration tests run the pipeline on a full-sized dataset on the AWS cloud infrastructure. This ensures that the pipeline runs on AWS, has sensible resource allocation defaults set to run on real-world datasets, and permits the persistent storage of results to benchmark between pipeline releases and other analysis sources. The results obtained from the full-sized test can be viewed on the [nf-core website](https://nf-co.re/hicar/results).
```

JoseEspinosa commented on conf/modules.config:
```diff
@@ -9,28 +9,20 @@
         mode: params.publish_dir_mode,
-        saveAs: { filename -> filename.equals('versions.yml') ? null : filename },
-        enabled: false
+        saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
```
Wouldn't it be better to enable it to `false` by default again and not need to declare some processes just to avoid the publication of its files (e.g. `GUNZIP`)?

JoseEspinosa commented on conf/test.config:
```diff
@@ -20,21 +20,24 @@ params {
-
-    // Output folder
-    outdir = 'test_results'
+    input = "${projectDir}/assets/test_multi_samplesheet.csv"
```
Should this not be available in [test-datasets](https://github.com/nf-core/test-datasets/tree/hicar) instead ?

edmundmiller replied:
I personally like to keep the test samplesheets in the repo as examples as well. That ensures they stay up to date! 😆

JoseEspinosa commented on conf/test_full.config:
```diff
@@ -17,9 +17,6 @@ params {
     // Input data for full size test
     input = "${projectDir}/assets/test_full_samplesheet.csv"
```
Same as my comment above

JoseEspinosa commented on conf/test_hipeak.config:
```diff
@@ -20,17 +20,17 @@ params {
     // Input data for full size test
     input = "${projectDir}/assets/test_multi_samplesheet.csv"
```
Same here, and any other `input` instance in the configs

JoseEspinosa commented on docs/usage.md:
```diff
@@ -61,40 +63,127 @@ TREATMENT,3,AEG588A6_S6_L004_R1_001.fastq.gz,AEG588A6_S6_L004_R2_001.fastq.gz,,
+On a large scale, the arrangement of chromosomes are organised into two compartments labelled A ("active") and B ("inactive").
+A/B compartment-associated regions are on the multi-Mb scale and correlate with either open and expression-active chromatin ("A" compartments) or closed and expression inactive chromatin ("B" compartments). A compartments tend to be gene-rich, have high GC-content, contain histone markers for active transcription, and usually displace the interior of the nucleus. The regions in compartment A tend ...
+
+A topologically associating domain (TAD) is a smaller size genomic region compared to A/B compartments. It is a self-interacting genomic region. Most of the studies indicate TADs regulate gene expression by limiting the enhancer-promoter interaction to each TAD. A number of proteins are known to be associated with TAD formation. The most studied proteins are the protein CTCF and the protein compl ...
```
Add the complete name of the protein the first time it name appears: `CCCTC-binding factor (CTCF)`

JoseEspinosa commented on modules/local/homer/install.nf:
```diff
@@ -0,0 +1,38 @@
+process HOMER_INSTALL {
```
Maybe call it `HOMER_CONFIGURE`? since it is more similar to the name of the script invoked

JoseEspinosa commented on modules/local/samtools/sort/main.nf:
```diff
@@ -2,16 +2,17 @@ process SAMTOOLS_SORT {
     tag "$meta.id"
```
I guess there is a reason to maintain a local version of samtools sort? Otherwise use the nf-core module

JoseEspinosa commented on subworkflows/local/apa.nf:
```diff
@@ -0,0 +1,69 @@
+    ch_multiqc_files        = Channel.empty() // TODO
+    ch_apa                  = Channel.empty() // a png files channel
+
+    switch(params.apa_tool){
```
Nice have never used a switch such as this one in a subworkflow 😍

JoseEspinosa commented on subworkflows/local/apa/cooltools.nf:
```diff
@@ -0,0 +1,24 @@
+ * call APA by cooltools
+ */
+
+include { COOLTOOLS_PILEUP                 } from '../../../modules/local/cooltools/pileup'
+include { PLOTNPZ_BY_COOLTOOLS                } from '../../../modules/local/cooltools/plotnpz'
```
```suggestion
include { COOLTOOLS_PILEUP     } from '../../../modules/local/cooltools/pileup'
include { PLOTNPZ_BY_COOLTOOLS } from '../../../modules/local/cooltools/plotnpz'
```

JoseEspinosa commented on subworkflows/local/apa/juicer.nf:
```diff
@@ -0,0 +1,33 @@
+    )
+    ch_version = JUICER_APA.out.versions
+
+
```
```suggestion
```

JoseEspinosa commented on subworkflows/local/apa/cooltools.nf:
```diff
@@ -0,0 +1,24 @@
+
+    PLOTNPZ_BY_COOLTOOLS(COOLTOOLS_PILEUP.out.npz, format)
+
+
```
```suggestion
```

JoseEspinosa commented on subworkflows/local/apa.nf:
```diff
@@ -0,0 +1,69 @@
+            break
+    }
+
+
```
```suggestion
```

JoseEspinosa commented on subworkflows/local/callatacpeak.nf:
```diff
@@ -93,8 +104,8 @@ workflow ATAC_PEAK {
     peak       = ch_group_peak                        // channel: [ val(meta), path(peak) ]
     mergedpeak = MERGE_PEAK.out.peak                  // channel: [ path(bed) ]
     stats      = ATACQC.out.stats                     // channel: [ path(csv) ]
-    reads      = DUMP_READS.out.peak                  // channel: [ val(meta), path(bedgraph) ]
-    samplereads= DUMP_READS.out.peak                  // channel: [ val(meta), path(bedgraph) ]
+    reads      = DUMP_READS_PER_GROUP.out.peak        // channel: [ val(meta), path(bedgraph) ]
+    samplereads= DUMP_READS_PER_SAMPLE.out.peak       // channel: [ val(meta), path(bedgraph) ]
     bws        = UCSC_BEDGRAPHTOBIGWIG_PER_GROUP.out.bigwig     // channel: [ val(meta), path(bigwig) ]
     versions   = ch_version                           // channel: [ path(version) ]
```
Align comments

JoseEspinosa commented on subworkflows/local/coveragescale.nf:
```diff
@@ -0,0 +1,20 @@
+    GET_SCALE(counts, min_cnt)
+
+    emit:
+    scale           = GET_SCALE.out.scale
```
Add channel description please

JoseEspinosa commented on subworkflows/local/differential_analysis.nf:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    emit:
+    diff            = ch_diff_files
+    circos          = ch_circos_files
+    igv             = ch_track_files
+    anno            = ch_annotation_files
+    versions        = ch_versions          // channel: [ versions.yml ]
+    mqc             = ch_multiqc_files
```
Add channels description please

JoseEspinosa commented on subworkflows/local/hipeak/call_hipeak.nf:
```diff
@@ -0,0 +1,88 @@
+    }
+
+    emit:
+    peak         = ASSIGN_TYPE.out.peak         // channel: [ path(peak) ]
+    bedpe        = ASSIGN_TYPE.out.bedpe        // channel: [ path(bedpe) ]
+    bed4tfea     = ch_bed                       // channel: [ meta, bed ]
+    stats        = stats                        // channel: [ path(stats) ]
+    diff         = diff                         // channel: [ path(diff) ]
+    versions     = ch_version                   // channel: [ path(version) ]
```
```suggestion
    peak     = ASSIGN_TYPE.out.peak         // channel: [ path(peak) ]
    bedpe    = ASSIGN_TYPE.out.bedpe        // channel: [ path(bedpe) ]
    bed4tfea = ch_bed                       // channel: [ meta, bed ]
    stats    = stats                        // channel: [ path(stats) ]
    diff     = diff                         // channel: [ path(diff) ]
    versions = ch_version                   // channel: [ path(version) ]
```

JoseEspinosa commented on subworkflows/local/interaction_caller/maps_peak.nf:
```diff
@@ -0,0 +1,57 @@
+    MAPS_STATS(MAPS_CALLPEAK.out.summary.map{it[2]}.collect())
+
+    emit:
+    peak         = MAPS_REFORMAT.out.bedpe      // channel: [ meta, bin_size, path(bedpe) ]
+    stats        = MAPS_STATS.out.summary       // channel: [ path(stats) ]
+    versions     = ch_version                   // channel: [ path(version) ]
```
```suggestion
    peak     = MAPS_REFORMAT.out.bedpe      // channel: [ meta, bin_size, path(bedpe) ]
    stats    = MAPS_STATS.out.summary       // channel: [ path(stats) ]
    versions = ch_version                   // channel: [ path(version) ]
```

JoseEspinosa commented on subworkflows/local/interactions.nf:
```diff
@@ -0,0 +1,96 @@
+    ch_versions = ch_versions.mix(MERGE_INTERACTIONS.out.versions)
+
+    emit:
+    loops           = ch_loops                              // channel: [ meta, bin_size, path(bedpe) ]
+    mergedloops     = MERGE_INTERACTIONS.out.interactions   // channel: [ bin_size, path(bedpe) ]
+    circos          = ch_circos_files
+    igv             = ch_track_files
+    anno            = ch_annotation_files
+    versions        = ch_versions                           // channel: [ versions.yml ]
+    mqc             = ch_multiqc_files
```
```suggestion
    loops       = ch_loops                              // channel: [ meta, bin_size, path(bedpe) ]
    mergedloops = MERGE_INTERACTIONS.out.interactions   // channel: [ bin_size, path(bedpe) ]
    circos      = ch_circos_files
    igv         = ch_track_files
    anno        = ch_annotation_files
    mqc         = ch_multiqc_files
    versions    = ch_versions                           // channel: [ versions.yml ]
```

JoseEspinosa commented on subworkflows/local/tads_caller/cooltools.nf:
```diff
@@ -0,0 +1,28 @@
+
+
+    emit:
+    tads         = ch_tads                      // channel: [ meta, bin_size, path(bedpe) ]
+    mqc          = ch_multiqc_files             // channel: [ path(mqc) ]
+    versions     = ch_versions                  // channel: [ path(version) ]
```
```suggestion
    tads     = ch_tads                      // channel: [ meta, bin_size, path(bedpe) ]
    mqc      = ch_multiqc_files             // channel: [ path(mqc) ]
    versions = ch_versions                  // channel: [ path(version) ]
```

JoseEspinosa commented on subworkflows/local/tads_caller/cooltools.nf:
```diff
@@ -0,0 +1,28 @@
+    ).tads
+    ch_versions = COOLTOOLS_INSULATION.out.versions
+
+
```
```suggestion
```

JoseEspinosa commented on subworkflows/local/tads_caller/homer.nf:
```diff
@@ -0,0 +1,31 @@
+        genome).tads
+    ch_versions = HOMER_FINDTADSANDLOOPS_TADS.out.versions
+
+
```
```suggestion
```

JoseEspinosa commented on subworkflows/local/tfea/homer.nf:
```diff
@@ -0,0 +1,36 @@
+    HOMER_FINDMOTIFSGENOME(ch_new_bed, additional_param)
+    ch_versions = ch_versions.mix(HOMER_FINDMOTIFSGENOME.out.versions)
+
+
```
```suggestion
```

JoseEspinosa commented on subworkflows/local/v4c.nf:
```diff
@@ -0,0 +1,59 @@
+            break
+    }
+
+
```
```suggestion
```

JoseEspinosa commented on subworkflows/local/v4c.nf:
```diff
@@ -0,0 +1,59 @@
+
+
+    emit:
+    versions        = ch_versions          // channel: [ versions.yml ]
```
Also here only version is emitted, is this correct?

JoseEspinosa commented on subworkflows/local/v4c/cooltools.nf:
```diff
@@ -0,0 +1,26 @@
+    ch_versions = ch_versions.mix(VIRTUAL4C_BY_COOLTOOLS.out.versions)
+
+    emit:
+    versions        = ch_versions          // channel: [ versions.yml ]
+    v4c             = VIRTUAL4C_BY_COOLTOOLS.out.v4c
```
```suggestion
    v4c      = VIRTUAL4C_BY_COOLTOOLS.out.v4c
    versions = ch_versions                    // channel: [ versions.yml ]
```

JoseEspinosa replied:
Also add channel content description

JoseEspinosa reviewed (changes requested):
Huge work @jianhong! 😄 
Apart from my suggestions in the code, I agree with @Emiller88 summary

jianhong committed:
fix a bug for pairsqcplot if no data is avalable.

jianhong committed:
Merge remote-tracking branch 'origin/dev2' into dev

jianhong committed:
update documentations.

jianhong committed:
fix some minor stuff for the review comments.

jianhong committed:
Merge branch 'nf-core:dev' into dev

jianhong committed:
Run prettier -w.

jianhong committed:
Merge pull request #96 from jianhong/dev

Fix multiple minor typos for review comments.

nf-core-bot committed:
WIP: Remove vulnerable PR-comment artifact pattern

Automated security rollout could not apply this cleanly. Resolve the
remaining conflict markers and *.rej files, then open a PR from 'patch'
into 'dev'.

Patch source: this pipeline's TEMPLATE branch never received the
nf-core/tools 4.0.3 sync, so the change was taken from
nf-core/abotyper's TEMPLATE with the repository identity in branch.yml
rewritten for this pipeline. While resolving, check that no file ended
up naming nf-core/abotyper.

mashehu committed:
Merge pull request #122 from nf-core/patch

Remove vulnerable PR-comment artifact pattern
